jwql/tests/test_utils.py

```diff
@@ -175,6 +177,35 @@ def test_filename_parser(filename, solution):
     assert filename_parser(filename) == solution
 
 
+@pytest.mark.xfail(reason='Known non-compliant files in filesystem')
+def test_filename_parser_whole_filesystem(capsys):
+    """Test the filename_parser on all files currently in the filesystem.
+    """
+    # Get all files
+    filesystem_dir = get_config()['filesystem']
+    all_files = []
+    for dirName, _, fileList in os.walk(filesystem_dir):
+        for file in fileList:
+            if file.endswith('.fits'):
+                all_files.append(os.path.join(dirName, file))
+
+    # Run the filename_parser on all files
```

Comment:
@laurenmarietta I think this is a good way to do this for now.  I do see this becoming potentially a problem when your filesystem gets to be huge -- this test might take forever to run.  But perhaps when it comes to that, we can implement `multiprocessing` or something to speed it up.

Reply:
Yes, I agree completely. This test already takes many times longer to run than the other tests in `test_utils`, I think just because of walking through the whole directory. So we should definitely keep an eye on it as things scale up!